acloud/data_source_node_join_config.go

```diff
@@ -0,0 +1,96 @@
+package acloud
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/avisi-cloud/go-client/pkg/acloudapi"
+)
+
+func dataSourceNodeJoinConfig() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceNodeJoinConfigRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"organisation_slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"node_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"user_data": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"kubelet_config": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"join_command": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"install_script": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"upgrade_script": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceNodeJoinConfigRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(acloudapi.Client)
+
+	organisationSlug := d.Get("organisation_slug").(string)
+	environmentSlug := d.Get("environment_slug").(string)
+	clusterSlug := d.Get("cluster_slug").(string)
+
+	cluster, err := client.GetCluster(ctx, organisationSlug, environmentSlug, clusterSlug)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if cluster == nil {
+		return diag.FromErr(fmt.Errorf("cluster was not found"))
+	}
+
+	nodePoolID := d.Get("node_pool_id").(string)
```

Comment:
Hoe komt men aan dit ID?

Reply:
This is exported from the node pool resource.

https://registry.terraform.io/providers/avisi-cloud/acloud/latest/docs/resources/nodepool#id